### PR TITLE
Update split header button active/disabled state styles

### PR DESCRIPTION
**Pull request by gretanausedaite**
- Removed .iui-active from child buttons in split header button
- Added style overrides to achieve same look
- Added disabled state styling using .iui-disabled and :disabled
- Removed some `svg.` selectors
- Increased mismatch threshold to fix random test failures

### CODE REVIEW

File: src/header/header.scss [hunk 1]
```diff
@@ -13,7 +13,8 @@
   user-select: none;
   box-sizing: border-box;
   // iPhone notch support
-  padding: env(safe-area-inset-top) env(safe-area-inset-right) 0 env(safe-area-inset-left);
+  padding: env(safe-area-inset-top) env(safe-area-inset-right) 0
```
Comment (bentleyvk) [inline src/header/header.scss:16]: I wonder why Greta PRs always fix formatting? @FlyersPh9

File: src/header/header.scss [hunk 2]
```diff
@@ -319,7 +341,6 @@
       }
     }
 
-    // Removing button hovers and backgrounds and adding last child hover
```
Comment (bentleyvk) [inline src/header/header.scss:322]: These comments was kind of useful. At least for me to read the code.

File: src/header/header.scss [hunk 3]
```diff
@@ -304,7 +305,28 @@
   }
 
   &.iui-active {
```
Comment (bentleyvk) [inline src/header/header.scss:307]: @FlyersPh9 @mayank99 We either make these changes or add active state to split button.
For me these changes seem better as they are small and easy to understand.

File: src/header/header.scss (hunk 1 above)
Comment (mayank99) [reply src/header/header.scss:16]: These formatting fixes are from Prettier and need to be done manually because we disabled autoformat (it was messing with Stylelint).

So Jon should always format before pushing.

File: src/header/header.scss (hunk 3 above)
Comment (mayank99) [reply src/header/header.scss:307]: Isn't there some mixin we can use?

Comment (mayank99) [reply src/header/header.scss:307]: Oh, there isn't. In that case, I think we should extract out the active state logic (in `button/default.scss`) into a mixin that we can just include here.

After that, we can override anything if needed.

File: src/header/header.scss [hunk 4]
```diff
@@ -291,6 +267,55 @@
   }
 }
 
+@mixin iui-header-button-active {
```
Comment (gretanausedaite) [inline src/header/header.scss:270]: @mayank99 Is this good enough? This doesn't break any other button styles and works for split button.

Comment (mayank99) [reply src/header/header.scss:270]: Can you move this mixin below split button? I can't even see the diff properly right now

Comment (mayank99) [reply src/header/header.scss:270]: Are you sure we need all these extra rules? I thought we would need fewer overrides after using the mixin.

Comment (mayank99) [reply src/header/header.scss:270]: Yeah, this is better. I did notice two different problems:

1. Focus styling for left button looks wrong (it should extend all the way to the left edge).
image

2. Disabled buttons still have hover effect and blue stripe.
image

File: src/header/header.scss [hunk 5]
```diff
@@ -341,5 +307,96 @@
         }
       }
     }
+
+    &.iui-disabled {
+      .iui-button {
```
Comment (mayank99) [inline src/header/header.scss:312]: I would suggest adding `.iui-header-button-icon { filter: grayscale(1) }` here

File: src/header/header.scss [hunk 6]
```diff
@@ -341,5 +307,96 @@
...
+    // Blue background and darker blue bottom line
+    @include themed {
+      box-shadow: 0 $iui-xxs 0 0 t(iui-color-foreground-primary);
+      background-color: rgba(
+        t(iui-color-foreground-primary-rgb),
+        t(iui-opacity-6)
+      );
+    }
+
+    svg.iui-header-button-icon {
```
Comment (mayank99) [inline src/header/header.scss:342]: `svg` should not be needed here

File: src/header/header.scss [hunk 7]
```diff
@@ -341,5 +307,96 @@
...
+      }
+    }
+
+    .iui-icon {
+      @include themed {
+        fill: t(iui-icons-color-actionable-disabled);
+      }
+    }
+
+    svg.iui-header-button-icon {
```
Comment (mayank99) [inline src/header/header.scss:395]: `svg` not needed here too

File: src/header/header.scss [hunk 8]
```diff
@@ -341,5 +307,96 @@
...
+      @include themed {
+        fill: t(iui-icons-color-actionable-disabled);
```
Comment (mayank99) [inline src/header/header.scss:397]: This can be removed as `.iui-icon` above does the same

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,15 @@
...
 
+## Unreleased
+
+`Date`
+
+### What's new
+
+### Fixes
+
+- **Fixed Split Header button active state styling** to require less class names.
```
Comment (mayank99) [inline CHANGELOG.md:11]: Grammar
```suggestion
- **Fixed Split Header button active state styling** to require fewer class names.
```

File: src/header/header.scss (hunk 5 above)
Comment (mayank99) [reply src/header/header.scss:312]: Actually, it should also be applied on normal header-button 🤔

File: backstop/tests/header.html [hunk 10]
```diff
@@ -752,7 +752,21 @@ <h2>Slim</h2>
 
     <hr />
 
-    <h2>Split Menu Buttons</h2>
+    <h2>
+      Split Menu Buttons
+      <label
```
Comment (mayank99) [inline backstop/tests/header.html:757]: This toggle switch will allow us to look at disabled state easily

File: src/header/header.scss [hunk 11]
```diff
@@ -341,5 +313,95 @@
...
+    }
+  }
+
+  // Disabled state
+  &.iui-disabled,
+  &:disabled {
+    @include themed {
+      background-color: rgba(t(iui-color-foreground-body-rgb), 0.05);
+      box-shadow: 0 $iui-xxs 0 0
+        rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-4));
```
Comment (mayank99) [inline src/header/header.scss:386]: These values are very random and may not be "correct". Lets decide the visuals once Jon gets back. I will create a discussion.

**Review by mayank99 — APPROVED** [review]
I removed some extra code and finished the PR so we can release and use this in React. Should be good to go now.

There is still a slight problem with focus styling, but it should be fixed in a future PR (or in #199).
image

Also we should revisit header button styling after #225. The nesting/specificity in here is a mess.

File: src/button/split-menu.scss [hunk 12]
```diff
@@ -44,4 +44,10 @@
       );
     }
   }
+
+  &.iui-disabled {
```
Comment (bentleyvk) [inline src/button/split-menu.scss:48]: What it gives?

File: backstop/scenarios/header.js [hunk 13]
```diff
@@ -4,27 +4,32 @@ module.exports = [
...
     actions: [hover('#test-button-3'), focus('#test-button-4')],
     selectors: ['#demo-slim'],
+    misMatchThreshold: 0.02,
   }),
 
   //// Split
   scenario('Type Split State hover & focus', {
     actions: [hover('#test-button-6'), focus('#test-button-6')],
     selectors: ['#demo-split'],
+    misMatchThreshold: 0.05,
```
[the remark this answers is not in the document]
Comment (bentleyvk) [reply backstop/scenarios/header.js:33]: I think this will be Chrome again rendering images a bit different every time. Because previously we had the same problem with `Tile` on hover where images are scaled up. I then made screenshot, refreshed, made another screenshot and compared them with image comparer. Turns out images always differs a bit in few pixels.

File: src/button/default.scss [hunk 14]
```diff
@@ -49,25 +49,29 @@
...
+    }
+  }
+}
 
-      > .iui-label {
-        @include themed {
-          color: t(iui-color-foreground-primary);
-        }
-      }
+@mixin iui-button-active {
```
Comment (bentleyvk) [inline src/button/default.scss:57]: I thought we will reuse this mixin inside header button but as I understand it really can't be reused there. So maybe undo this change.

**Review by bentleyvk — APPROVED** [review]
